src/Eshopworld.WorkerProcess/Configuration/ServiceConfigurationExtensions.cs

```diff
@@ -33,5 +40,29 @@ public static void AddWorkerLease(this IServiceCollection services)
 
             services.TryAddSingleton<IWorkerLease, WorkerLease>();
         }
+
+        /// <summary>
+        /// Adds <see cref="DistributedLock"/> to the services collection
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="cosmosDataStoreOptions"></param>
+        /// <param name="cosmosDbConnectionOptions"></param>
+        public static void AddCosmosDistributedLock(this ContainerBuilder builder, CosmosDataStoreOptions cosmosDataStoreOptions, CosmosDbConnectionOptions cosmosDbConnectionOptions)
+        {
+            builder.Register(ctx => Options.Create(cosmosDataStoreOptions))
+                .As<IOptions<CosmosDataStoreOptions>>().SingleInstance();
+
+            builder.Register(
+                ctx => new DocumentClient(cosmosDbConnectionOptions.ConnectionUri, cosmosDbConnectionOptions.AccessKey)
+            ).Named<IDocumentClient>("WorkerProcessDocumentClient");
```

Comment:
can this be moved to a constant?

Reply:
I believe that you are talking about the string. So yes, doing it now.